src/openttd.cpp

```diff
@@ -940,7 +941,17 @@ bool SafeLoad(const std::string &filename, SaveLoadOperation fop, DetailedFileTy
 	_game_mode = newgm;
 
 	switch (lf == nullptr ? SaveOrLoad(filename, fop, dft, subdir) : LoadWithFilter(lf)) {
-		case SL_OK: return true;
+		case SL_OK:
+			/* Start the GameScript. */
+			Game::StartNew();
+
+			/* Start the AIs. */
+			for (const Company *c : Company::Iterate()) {
+				if (Company::IsValidAiID(c->index)) AI::StartNew(c->index, false);
+			}
+
+			ShowAIDebugWindowIfAIError();
```

Comment:
I think this should be a separate function named `StartScripts()` or so. Maybe even in some file in the script folder.